hnn_core/network.py

```diff
@@ -733,21 +736,15 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
             # allows tests must match HNN GUI output by preserving original
             # gid assignment convention
             target_populations = list(self.cell_types.keys())
+            for target_type in target_populations:
+                if target_type not in weights_by_type:
+                    weights_by_type.update({target_type: {'ampa': 0.}})
+                if target_type not in delays_by_type:
+                    delays_by_type.update({target_type: 0.1})
         elif len(target_populations) == 0:
             raise ValueError('No target populations have been specified for '
                              'this drive.')
 
-        weights_by_receptor = {'ampa': weights_ampa, 'nmda': weights_nmda}
-        if isinstance(synaptic_delays, dict):
-            for receptor in ['ampa', 'nmda']:
-                # synaptic_delays must be defined for all cell types for which
-                # either AMPA or NMDA weights are non-zero
-                if not (set(weights_by_receptor[receptor].keys()).issubset(
-                        set(synaptic_delays.keys()))):
-                    raise ValueError(
-                        'synaptic_delays is either a common float or needs '
-                        'to be specified as a dict for each cell type')
-
```

Comment:
This, along with other new checks that pertain to synaptic weights and delays have been moved to `drives._get_target_properties()`